remapping_commands.sh

```diff
@@ -0,0 +1,181 @@
+bedtools flank -i variants.bed -g "$oldgenome".chrom.sizes -b 50 > flanking.bed
+rm variants.bed "$oldgenome".chrom.sizes
+
+#this outputs a flanking.bed file: for 1 variant we have the following lines:
+#[chr] [-50 pos] [variant pos-1] [rsID] [qual] [REF allele] [ALT allele] [filter] [info]
+#[chr] [variant pos] [+50 pos] [rsID] [qual] [REF allele] [ALT allele] [filter] [info]
+
+#we want 1 line per variant, with 1 interval to create the read:
+#[chr] [-50 pos] [+50 pos] [rsID] [qual] [REF allele] [ALT allele] [filter] [info]
+
+#fix rows: put both flanking sequences on 1 row so each read is 101 bases: [50 bases]-[variant]-[50 bases]
+awk 'NR == 1 {prev = $2} NR>1 && NR%2==0 {print $1"\t"prev"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7"\t"$8"\t"$9} NR%2 == 1 { prev = $2}' flanking.bed > flanking.corrected.bed
+rm flanking.bed
+
+#get the fasta sequences for these intervals
+bedtools getfasta -fi $oldgenome -bed flanking.corrected.bed -fo variants_reads.fa
+
+#replace the colon separators with "|"
+sed -i 's/:/|/' variants_reads.fa
+
+#store rsIDs
+awk '{print $4}' flanking.corrected.bed > rsIDs.txt
+
+#store variant bases
+awk '{print $7}' flanking.corrected.bed > variant_bases.txt
+
+#store the other vcf columns
+awk '{print $5, $8, $9}' flanking.corrected.bed > qual_filt_info.txt
+
+#paste the names, variant bases, then fasta sequences into a new file
+paste <(grep '^>' variants_reads.fa) variant_bases.txt rsIDs.txt qual_filt_info.txt <(grep -v '^>' variants_reads.fa) > temp.txt
+rm flanking.corrected.bed variant_bases.txt rsIDs.txt qual_filt_info.txt  variants_reads.fa
+
+#reformat the fasta ID
+sed 's/\t/|/g; s/ /|/g; s/\(.*\)|/\1 /' temp.txt | tr ' ' '\n' > variant_reads.out.fa
+rm temp.txt
+
+
+
+
+
+
+#2) Mapping with bowtie2
+echo "---------------------------2) Mapping with bowtie2---------------------------"
+#"index" is the prefix of the output filenames, not a directory
+echo "#### Building index: may take a while ####"
+bowtie2-build $newgenome index
+echo "#### Mapping results: ####"
+bowtie2 -f -x index variant_reads.out.fa | samtools view -bS - > reads_aligned.bam
+#example output:
+#>1094 reads; of these:
+#>  1094 (100.00%) were unpaired; of these:
+#>   7 (0.64%) aligned 0 times
+#>    48 (4.39%) aligned exactly 1 time
+#>    1039 (94.97%) aligned >1 times
+#>99.36% overall alignment rate
+
+
+
+
+
+#3) Data extraction
+echo '------------------------------3) Data extraction-----------------------------'
+#extract chromosome, rsID, position on new genome, original variant allele, qual, filter and info fields
+samtools sort reads_aligned.bam -o reads_aligned.sorted.bam
+samtools index reads_aligned.sorted.bam
+./reverse_strand.py -i reads_aligned.sorted.bam -o variants_remapped.vcf
+rm reads_aligned.bam reads_aligned.sorted.bam reads_aligned.sorted.bam.bai
+
+#add interval for variant position in bed format (required by getfasta)
+awk '{print $1"\t"$2-1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7}' variants_remapped.vcf > variants_remapped.bed
```

Comment:
Again this is a bit obscure, could you please provide an input-output example here?